scheduler/plugin/wrappedplugin.go

```diff
@@ -0,0 +1,270 @@
+	}
+
+	fp, ok := p.(framework.FilterPlugin)
+	if ok {
+		plg.originalFilterPlugin = fp
+	}
+
+	sp, ok := p.(framework.ScorePlugin)
+	if ok {
+		plg.originalScorePlugin = sp
+	}
+
+	return plg
+}
+
+func (w *wrappedPlugin) Name() string { return w.name }
+func (w *wrappedPlugin) ScoreExtensions() framework.ScoreExtensions {
+	if w.originalScorePlugin != nil && w.originalScorePlugin.ScoreExtensions() != nil {
+		return w
+	}
+	return nil
+}
+
+// NormalizeScore wraps original NormalizeScore plugin of Scheduler Framework.
+// Before and after the execution of original NormalizeScore plugin,
+// we will run arbitrary processing as functions from normalizeScorePluginExtender.
+func (w *wrappedPlugin) NormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	if w.originalScorePlugin == nil || w.originalScorePlugin.ScoreExtensions() == nil {
+		// return nil not to affect scoring
+		return nil
+	}
+
+	if w.normalizeScorePluginExtender != nil {
+		if s := w.doBeforeNormalizeScoreAndStoreResult(ctx, state, pod, scores); !s.IsSuccess() {
+			return s
+		}
+	}
+	orignstatus := w.originalScorePlugin.ScoreExtensions().NormalizeScore(ctx, state, pod, scores)
+	if !orignstatus.IsSuccess() {
+		klog.Errorf("failed to run normalize score: %v, %v", orignstatus.Code(), orignstatus.Message())
+		// If it is not nil, we wiil return the results of AfterNormalizeScore.
+		if w.normalizeScorePluginExtender != nil {
+			return w.doAfterNormalizeScoreAndStoreResult(ctx, state, pod, scores, orignstatus)
+		}
+		// Otherwise, we will return the original results.
+		return orignstatus
+	}
+
+	for _, s := range scores {
+		w.store.AddNormalizedScoreResult(pod.Namespace, pod.Name, s.Name, w.originalScorePlugin.Name(), s.Score)
+	}
+	// If it is not nil, we wiil run AfterNormalizeScore.
+	if w.normalizeScorePluginExtender != nil {
+		_ = w.doAfterNormalizeScoreAndStoreResult(ctx, state, pod, scores, orignstatus)
+	}
+
+	return nil
+}
+
+func (w *wrappedPlugin) doBeforeNormalizeScoreAndStoreResult(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
+	s := w.normalizeScorePluginExtender.BeforeNormalizeScore(ctx, state, pod, scores)
+	for _, score := range scores {
+		w.store.AddNormalizedScoreResult(pod.Namespace, pod.Name, score.Name, resultBeforePrefix+w.originalScorePlugin.Name(), score.Score)
+	}
+	return s
+}
+
+func (w *wrappedPlugin) doAfterNormalizeScoreAndStoreResult(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList, orignstatus *framework.Status) *framework.Status {
+	s := w.normalizeScorePluginExtender.AfterNormalizeScore(ctx, state, pod, scores, orignstatus)
+	for _, score := range scores {
+		w.store.AddNormalizedScoreResult(pod.Namespace, pod.Name, score.Name, resultAfterPrefix+w.originalScorePlugin.Name(), score.Score)
+	}
```

Comment:
duplicated with L170-L172, so not needed.

Reply:
Hmm, so I think these doBeforeXXX and doAfterXXX are not needed.

Reply:
It will occur some lint error if it isn't split for another function.
For example, in `Score` function.
> : calculated cyclomatic complexity for function NormalizeScore is 11, max is 10 (cyclop)

Reply:
np. If it is better to ignore lint for readability or other reasons, we should ignore it.